File: src/api/timer.js
import apiClient from "./client.js";

// 타이머 페이지 - README.md "7. 타이머 페이지".

export function getTimer() {
  // GET /timer (인증 없음)
  // { name, status, start_time, end_time, time_until_start,
  //   remaining_seconds, remaining_display, server_time }.
  // status: BEFORE/RUNNING/ENDED. 활성 대회 없음 -> 200 + data: null
  // (404 아님, README 0-2절). server_time은 2026-09-03 Notion API명세서
  // 갱신으로 추가됨(README Appendix B #11 해소) - board/dice/status와
  // 동일하게 이 값으로 클라이언트-서버 시계 오차를 보정한다(timerData.js).
  return apiClient.get("/timer");
}

File: mock-backend/api/views/timer.py
from django.utils import timezone
from rest_framework.permissions import AllowAny
from rest_framework.views import APIView

from ..models import ContestTimer
from ..response import success


class TimerView(APIView):
    permission_classes = [AllowAny]

    def get(self, request):
        contest = ContestTimer.objects.order_by("-id").first()
        if contest is None:
            # README 0-2절: 활성 대회 없음 -> 200 + data: null (404 아님)
            return success(None)

        now = timezone.now()
        if now < contest.start_time:
            status_value = "BEFORE"
        elif now < contest.end_time:
            status_value = "RUNNING"
        else:
            status_value = "ENDED"

        remaining = max(0, int((contest.end_time - now).total_seconds())) if status_value == "RUNNING" else 0
        until_start = max(0, int((contest.start_time - now).total_seconds())) if status_value == "BEFORE" else 0

        hours, rem = divmod(remaining, 3600)
        minutes, seconds = divmod(rem, 60)

        return success(
            {
                "name": contest.name,
                "status": status_value,
                "start_time": contest.start_time,
                "end_time": contest.end_time,
                "time_until_start": until_start,
                "remaining_seconds": remaining,
                "remaining_display": f"{hours:02d}:{minutes:02d}:{seconds:02d}",
            }
        )
